pkg/daemon/daemon.go

```diff
@@ -660,7 +660,7 @@ func (dn *Daemon) getStateAndConfigs(pendingConfigName string) (*stateAndConfigs
 		return nil, err
 	}
 	desiredConfigName, err := getNodeAnnotation(dn.node, constants.DesiredMachineConfigAnnotationKey)
-	if err != nil {
+	if err != nil && pendingConfigName == "" {
```

Comment:
I don't follow the logic here; we get an error finding `desiredConfigName` but then we're conditionalizing on the `pendingConfigName`?

Reply:
I need to drop this. Not sure why I've added it at this point.